sqlglot/generator.py

```diff
@@ -368,15 +368,18 @@ def create_sql(self, expression):
         expression_sql = self.sql(expression, "expression")
         expression_sql = f"AS{self.sep()}{expression_sql}" if expression_sql else ""
         temporary = " TEMPORARY" if expression.args.get("temporary") else ""
+        transient = (
```

Comment:
add CREATE_TRANSIENT to the base generator constants

then this is just

if self.CREATE_TRANSIENT and expression.args.get("transient")